### PR TITLE
sample: Basic Thread Demo

### PR DESCRIPTION
**Pull request by ric96**

### CODE REVIEW

File: samples/basic/threads/README.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Each thread is then defined at compile time using K_THREAD_DEFINE.
+
+Building
+********
+
+.. code-block:: console
+
+    $ cd samples/basic/threads
+    $ make
```
Comment (dbkinder) [inline samples/basic/threads/README.rst]: Nice start... how do you run the sample?  Will it only run on a 96b_carbon (worth mentioning if so).

File: samples/basic/threads/src/main.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+void uart_out(void)
+{
+	int cnt = 1;
+
+	while (1) {
+		printk("Toggle USR1 LED: Counter = %d\n", cnt);
+		if (cnt >= 10) {
+			printk("Toggle USR2 LED: Counter = %d\n", cnt);
+			cnt = 0;
```
Comment (dbkinder) [inline samples/basic/threads/src/main.c]: should this be ``cnt =1;`` to match what it was initialized to outside the loop?

**Review by dbkinder — CHANGES_REQUESTED** [review]
Great start, thanks.

File: samples/basic/threads/README.rst (hunk 1 above)
Comment (ric96) [reply samples/basic/threads/README.rst]: Like other samples, you would need to change the pin numbering for other boards, I'll mention it.

File: samples/basic/threads/src/main.c (hunk 2 above)
Comment (ric96) [reply samples/basic/threads/src/main.c]: there is an increment for cnt right before the loop ends, If I have cnt=1 here the next loop starts with cnt=2

File: samples/basic/threads/README.rst (hunk 1 above)
Comment (pfalcon) [reply samples/basic/threads/README.rst]: @ric96, Thanks for coming up with the updated version. And nope, not really, user doesn't need to change anything, the sample just needs to use <board.h> and LED0_GPIO_PORT, LED0_GPIO_PIN, etc. it defines. See basic/blinky for an example of writing portable hw-related sample.

Comment (pfalcon) [reply samples/basic/threads/README.rst]: @ric96 : Ping. Is the above gives enough info or more details would be helpful?

File: samples/basic/threads/src/main.c (hunk 2 above)
Comment (dbkinder) [reply samples/basic/threads/src/main.c]: ah, ok.

File: samples/basic/threads/README.rst (hunk 1 above)
Comment (ric96) [reply samples/basic/threads/README.rst]: @pfalcon seems enough to me, I've covered the "what it does" part under Overview. And i do not want to include the "how to flash" part as it varies from board to board.

Comment (pfalcon) [reply samples/basic/threads/README.rst]: Oops, I missed that you've already rewritten it to use board.h, that's what I meant. Will try it tomorrow.

**Review by dbkinder — APPROVED** [review]

**Review by pfalcon — APPROVED** [review]
Tested with https://github.com/zephyrproject-rtos/zephyr/pull/4756 on frdm_kw41z.

**Review by nashif — APPROVED** [review]

**Pull request merged**